### ISSUE TITLE
Percolator query not returning document that the query matches (multiple must nested conditions)

### ISSUE THREAD

**Issue by maricn** [elastic/elasticsearch#28911]
**Elasticsearch version** (`bin/elasticsearch --version`):
```
Version: 6.2.2, Build: 10b1edd/2018-02-16T19:01:30.685723Z, JVM: 9.0.4
```

**Plugins installed**: []

**JVM version** (`java -version`):
```
java version "9.0.4"
Java(TM) SE Runtime Environment (build 9.0.4+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.4+11, mixed mode)
```

**OS version** (`uname -a` if on a Unix-like system):
```
Darwin anotherMBP.local 17.4.0 Darwin Kernel Version 17.4.0: Sun Dec 17 09:19:54 PST 2017; root:xnu-4570.41.2~1/RELEASE_X86_64 x86_64
```

**Description of the problem including expected versus actual behavior**:

Percolator queries are not matching the documents that matched using the same queries.

I will include a query and a document below - when I index the document and then run the provided query against it, the document is found. When I add the same query to a percolator document and query the user against it, it is not found.

Related to https://github.com/elastic/elasticsearch/issues/28315.

I noticed this behavior only when the query has `must: []` containing multiple `nested` queries, having at least one of them containing multiple `must: []` conditions.

**Steps to reproduce**:

 1. I've just noticed this bug with must condition. I'm executing the following query: `GET /user/_search`
```
{
  "query": {
    "bool": {
      "must": [
        {
          "nested": {
            "query": {
              "bool": {
                "must": [
                  {
                    "term": {
                      "stories.storyId": 2
                    }
                  }
                ]
              }
            },
            "path": "stories"
          }
        },
        {
          "nested": {
            "query": {
              "bool": {
                "must": [
                  {
                    "term": {
                      "stories.storyId": 1
                    }
                  },
                  {
                    "term": {
                      "stories.storyName": "story1"
                    }
                  }
                ]
              }
            },
            "path": "stories"
          }
        },
        {
          "nested": {
            "query": {
              "bool": {
                "must": [
                  {
                    "range": {
                      "stories.executedAt": {
                        "gte": "now-1w"
                      }
                    }
                  }
                ]
              }
            },
            "path": "stories"
          }
        }
      ]
    }
  }
}
```
2. I get back the response with the document: 
```
{
  "took": 3,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 1,
    "max_score": 4.568616,
    "hits": [
      {
        "_index": "user-180306",
        "_type": "user",
        "_id": "kpF7-mEBH3oCPpbkErLB",
        "_score": 4.568616,
        "_source": {
          "stories": [
            {
              "storyId": 1,
              "storyName": "story1",
              "executedAt": "2018-03-06T07:45:00.000+01:00"
            },
            {
              "storyId": 2,
              "storyName": "story2",
              "executedAt": "2018-03-06T07:55:00.000+01:00"
            }
          ]
        }
      }
    ]
  }
}
```
 3. I percolate that search: `POST /story-query/doctype/`
```
{
          "query": {
            "bool": {
              "must": [
                {
                  "nested": {
                    "query": {
                      "bool": {
                        "must": [
                          {
                            "term": {
                              "stories.storyId": 2
                            }
                          }
                        ]
                      }
                    },
                    "path": "stories"
                  }
                },
                {
                  "nested": {
                    "query": {
                      "bool": {
                        "must": [
                          {
                            "term": {
                              "stories.storyId": 1
                            }
                          },
                          {
                            "term": {
                              "stories.storyName": "story1"
                            }
                          }
                        ]
                      }
                    },
                    "path": "stories"
                  }
                },
                {
                  "nested": {
                    "query": {
                      "bool": {
                        "must": [
                          {
                            "range": {
                              "stories.executedAt": {
                                "gt": "now-1w"
                              }
                            }
                          }
                        ]
                      }
                    },
                    "path": "stories"
                  }
                }
              ]
            }
          },
          "meta": {
            "storyId": 303,
            "storyName": "story-last-week-name",
            "storyDescription": "some story happened inside the last week, multiple conditions - name"
          }
        }
```
 4. When I run that document from result against the percolator, I don't get the percolated query in result list (i get other valid queries). Running document against percolator `GET story-query/doctype/_search`:
```
{
  "query": {
    "percolate": {
      "field": "query",
      "document_type": "doctype",
      "index": "user-180306",
      "type": "user",
      "id": "kpF7-mEBH3oCPpbkErLB"
    }
  },
  "_source": {
    "includes": "meta.*"
  }
}
```

**Provide logs (if relevant)**:

**Comment by javanna** [comment]
cc @elastic/es-search-aggs maybe @martijnvg would like to have a look?

**Comment by maricn** [comment]
Any help would be great, this is currently a blocker for us using ES. It would be nice to know if it's indeed a bug and if so, when could we expect it fixed. Any additional information or a workaround would be appreciated. Thanks!

**Comment by martijnvg** [comment]
@maricn Thanks for reporting. This is indeed a bug and should be fixed via #28840.

**Comment by martijnvg** [comment]
I just verified and #28840 does fix the problem here.

**Comment by martijnvg** [comment]
Fixed via #28840
